src/proxy/main.c

```diff
@@ -0,0 +1,118 @@
+#include <errno.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "libhirte/bus/bus.h"
+#include "libhirte/common/common.h"
+#include "libhirte/common/opt.h"
+#include "libhirte/log/log.h"
+
+const struct option options[] = { { ARG_HELP, no_argument, 0, ARG_HELP_SHORT }, { NULL, 0, 0, '\0' } };
+
+#define OPTIONS_STR ARG_HELP_SHORT_S
+
+static const char *opt_node_unit = NULL;
+
+static void usage(char *argv[]) {
+        printf("Usage: %s \n"
+               "\t[-h help]\t\t Print this help message.\n"
+               "\t<node>_<service>\t Input the name of the node the required service is supposed to run on.\n",
+               argv[0]);
+}
+
+int parse_node_unit_opt(const char *opt_node_unit, UNUSED char **ret_node_name, UNUSED char **ret_unit_name) {
+        if (opt_node_unit == NULL || ret_node_name == NULL || ret_unit_name == NULL) {
+                return -EINVAL;
+        }
+
+        // copy the option string as strtok_r manipulates it
+        char *rest = strdup(opt_node_unit);
+        if (rest == NULL) {
+                return -ENOMEM;
+        }
+
+        _cleanup_free_ char *token = NULL;
+        token = strtok_r(rest, "__", &rest);
```

Comment:
This feels a bit weird to me. I would rather pass in the nodename and the service name as separate options and then combine it however is needed internally rather than encode some way to combine them into one in the CLI interface.

Reply:
I agree. This would be clearer to the user of the proxy binary. 
The `<node>__<service>` pattern I took from #95 where we use the service like `service@nodename_servicename.service` and pass it to the proxy via `%i`. Not sure how we could split it in the serivce file.

Reply:
Ah, I see. Yeah, might be hard to do that then...